python/nav/mibs/juniper_mib.py

```diff
@@ -15,24 +15,43 @@
 #
 """JUNIPER-MIB MibRetriever"""
 from twisted.internet import defer
+
+from nav.oids import OID
 from nav.smidumps import get_mib
 from nav.mibs.mibretriever import MibRetriever
+from nav.models.manage import PowerSupplyOrFan as FRU
+from nav.ipdevpoll.shadows import PowerSupplyOrFan, Device
+
+
+OPERATING_DESCR = "jnxOperatingDescr"
+OPERATING_CPU = "jnxOperatingCPU"
+LOAD_AVG_1MIN = "jnxOperating1MinLoadAvg"
+LOAD_AVG_5MIN = "jnxOperating1MinLoadAvg"
+LOAD_AVG_15MIN = "jnxOperating1MinLoadAvg"
 
-OPERATING_DESCR = 'jnxOperatingDescr'
-OPERATING_CPU = 'jnxOperatingCPU'
-LOAD_AVG_1MIN = 'jnxOperating1MinLoadAvg'
-LOAD_AVG_5MIN = 'jnxOperating1MinLoadAvg'
-LOAD_AVG_15MIN = 'jnxOperating1MinLoadAvg'
+FRU_STATUS_MAP = {
+    "unknown": FRU.STATE_UNKNOWN,
+    "empty": FRU.STATE_UNKNOWN,
+    "present": FRU.STATE_UNKNOWN,
+    "ready": FRU.STATE_UNKNOWN,
+    "announceOnline": FRU.STATE_UNKNOWN,
+    "online": FRU.STATE_UP,
+    "anounceOffline": FRU.STATE_WARNING,
+    "offline": FRU.STATE_DOWN,
+    "diagnostic": FRU.STATE_WARNING,
+    "standby": FRU.STATE_WARNING,
+}
```

Comment:
Specifically, it would be nice with some feedback on whether these mappings of FRU states to NAV states (unknown/up/down/warning) make sense.

Reply:
Well, I had independent comments from two separate Juniper device operators now, and can't point out anything wrong with this, so I'm leaving it as is.